File: app/src/main/java/com/example/queenb/ExpandableTextViewAdapter.java
package com.example.queenb;


import android.content.Context;
import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.webkit.WebView;
import android.widget.BaseExpandableListAdapter;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ExpandableListView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class ExpandableTextViewAdapter extends BaseExpandableListAdapter {
    Context context;
    String QUESTION1 = "מי אנחנו?";
    String QUESTION2 = "מה משמעות השם QuennB?";
    String QUESTION3 = "איפה מתקיימת הפעילות של העמותה?";
    String QUESTION4 = "איך אני נרשמת לתכנית?";
    String QUESTION5 = "איך תתקיים הפעילות אם יהיה סגר?";

    String ANSWER1 = "היא עמותה שהקומה על ידי סטודנטיות למדעי המחשב מהאוניברסיטה העברית לאור מיעוט הנשים סביבן," +
            "  במטרה להעלות את מספר הנשים הבוחרות לעסוק במקצועות טכנולוגיים" +
            "  ובכך לשמר את מעמדה של מדינת ישראל כמעצמת הייטק.";
    String ANSWER2 = "משמעות אחת נובעת מהביטוי queen bee, מלכת הדבורים - נקבה חזקה בטבע וביטוי שגור" +
            " בשפה האנגלית לבחורה חזקה ומובילה. משמעות נוספת נובעת מהמילה be, היה לנו חשוב דרך השם להעביר את החשיבות שאנחנו רואות בכך שתהיי עצמך - תגשימי את עצמך " +
            "בצורה מלאה ותגדירי את עצמך ללא תלות בתכתיבי החברה. כמובן שהקריצה לביונסה לא מקרית.";
    String ANSWER3 = "העמותה פועלת בקרב אוכלוסיית הסטודנטיות למדעי המחשב בשישה מוקדים בארץ:\n" +
            "-אוניברסיטת תל אביב\n" +
            "-האוניברסיטה העברית\n" +
            "-הטכניון בחיפה\n" +
            "-אוניברסיטת בן גוריון בבאר שבע\n" +
            "-בחרת Western digital בכפר סבא\n" +
            "-JestHUB";
    String ANSWER4 = "מוזמנת להרשם בדף זה ונהיה בקשר בקרוב.";
    String ANSWER5 = "אם לא יהיה סגר, ניפגש במוקדים ברחבי הארץ. במקרה של סגר נעבור לזום.";



    String[] questions = {QUESTION1, QUESTION2, QUESTION3, QUESTION4, QUESTION5};

    String[][] answers = {{ANSWER1},{ANSWER2}, {ANSWER3}, {ANSWER4}, {ANSWER5}};


    public ExpandableTextViewAdapter(Context context) {
        this.context = context;
    }

    @Override
    public int getGroupCount() {
        return questions.length;
    }

    @Override
    public int getChildrenCount(int groupPosition) {
        return answers[groupPosition].length;
    }

    @Override
    public Object getGroup(int groupPosition) {
        return questions[groupPosition];
    }

    @Override
    public Object getChild(int groupPosition, int childPosition) {
        return answers[groupPosition][childPosition];
    }

    @Override
    public long getGroupId(int groupPosition) {
        return groupPosition;
    }

    @Override
    public long getChildId(int groupPosition, int childPosition) {
        return childPosition;
    }

    @Override
    public boolean hasStableIds() {
        return false;
    }

    @Override
    public View getGroupView(int groupPosition, boolean isExpanded, View convertView, ViewGroup parent) {
        String question = (String) getGroup(groupPosition);
        if(convertView==null){
            LayoutInflater inflater = (LayoutInflater)this.context.getSystemService(context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.faq_title, null);

        }

        TextView question2 = convertView.findViewById(R.id.feqTitleView);
        question2.setTypeface(null, Typeface.BOLD);
        question2.setText(question);
        return convertView;
    }

    @Override
    public View getChildView(int groupPosition, int childPosition, boolean isLastChild, View convertView, ViewGroup parent) {
        final String answer= (String) getChild(groupPosition, childPosition);

        if(convertView==null){
            LayoutInflater inflater = (LayoutInflater)this.context.getSystemService(context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.faq_answers, null);

        }

        TextView answer2 = convertView.findViewById(R.id.descriptionFaqView);
        answer2.setText(answer);
        return convertView;
    }

    @Override
    public boolean isChildSelectable(int groupPosition, int childPosition) {
        return false;
    }
}

File: app/src/main/java/com/example/queenb/ExpandableTextViewAdapter.kt
package com.example.queenb

import android.content.Context
import android.graphics.Typeface
import android.text.SpannedString
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseExpandableListAdapter
import android.widget.TextView
import androidx.core.content.res.TypedArrayUtils.getText

class ExpandableTextViewAdapter(var context: Context) :

    BaseExpandableListAdapter() {

    var QUESTION1 = "מי אנחנו?"
    var QUESTION2 = "מה משמעות השם QueenB?"
    var QUESTION3 = "איפה מתקיימת הפעילות?"
    var QUESTION4 = "איך מתקיימת הפעילות בסגר?"
    var QUESTION5 = "איך אני נרשמת לתכנית?"
    var QUESTION6 = "למה כדאי לנערה בת 14 ללמוד לתכנת?"

    var ANSWER1 =
        "העמותה QueenB הוקמה על ידי סטודנטיות למדעי המחשב מהאוניברסיטה העברית לאור מיעוט הנשים סביבן," +
                " במטרה להעלות את מספר הנשים הבוחרות לעסוק במקצועות טכנולוגיים" +
                "  ובכך לשמר את מעמדה של מדינת ישראל כמעצמת הייטק."
    var ANSWER2 =
        "משמעות אחת נובעת מהביטוי queen bee, מלכת הדבורים - נקבה חזקה בטבע וביטוי שגור" +
                " בשפה האנגלית לבחורה חזקה ומובילה. משמעות נוספת נובעת מהמילה be, היה לנו חשוב דרך השם להעביר את החשיבות שאנחנו רואות בכך שתהיי עצמך - תגשימי את עצמך " +
                "בצורה מלאה ותגדירי את עצמך ללא תלות בתכתיבי החברה. כמובן שהקריצה לביונסה לא מקרית."

    var ANSWER6 =
                "ענף ההייטק הוא הענף היציב והרווחי ביותר במשק הישראלי, במיוחד היום. חברות ההייטק מחפשות עוד ועוד כח אדם מתאים ומבינות את החשיבות שבגיוון אנושי. לימוד תכנות היא נקודת פתיחה שיכולה להבטיח עצמאות כלכלית.\n" +
                "\n" +
                "תכנות הוא לא רק כלי שמבטיח עתיד מקצועי. ביחד אנחנו לומדות להיות בטוחות יותר בעצמנו, להיות אמיצות להתנסות בתחומים חדשים, מחזקות את עצמנו ואחת את השנייה."

    var ANSWER3 =
        "העמותה פועלת בקרב אוכלוסיית הסטודנטיות למדעי המחשב ברחבי הארץ:\n" +
                "-תל אביב\n" +
                "-ירושלים\n" +
                "-בחיפה\n" +
                "-באר שבע\n" +
                "-מודיעין\n" +
                "-הרצליה\n" +
                "-הרצליה\n" +
                "לרשימה המפורטת של המוקדים והרשמה היכנסי לכאן https://www.queenb.org.il/signup"
    var ANSWER4 = "אנחנו ממשיכות גם באונליין!\n בשלב זה אנחנו נערכות לפתיחת השנה עם נוכחות פיזית במוקדי הפעילות שלנו, אך המצב עשוי להשתנות בכל עת בהתאם להנחיות משרדי החינוך והבריאות.\n" +
            "\n" +
            "במקרה כזה הפעילות שלנו תעבור במתכונת היברידית, המשלבת מפגשי תכנות אונליין ומפגשים קבוצתיים באוויר הפתוח. אל תדאגו, יש לנו כבר ארבעה חודשי ניסיון ואת כל מה שדרוש כדי להעביר את הפעילות שלנו בצורה הכי איכותית, שלמה ומהנה גם בצל ההגבלות."
    var ANSWER5 = "מוזמנת להרשם בלינק הבא https://www.queenb.org.il/signup ונהיה בקשר בקרוב."

    var questions =
        arrayOf(QUESTION1, QUESTION2, QUESTION6, QUESTION3, QUESTION4, QUESTION5)
    var answers = arrayOf(
        arrayOf(ANSWER1),
        arrayOf(ANSWER2),
        arrayOf(ANSWER6),
        arrayOf(ANSWER3),
        arrayOf(ANSWER4),
        arrayOf(ANSWER5)
    )

    override fun getGroupCount(): Int {
        return questions.size
    }

    override fun getChildrenCount(groupPosition: Int): Int {
        return answers[groupPosition].size
    }

    override fun getGroup(groupPosition: Int): Any {
        return questions[groupPosition]
    }

    override fun getChild(groupPosition: Int, childPosition: Int): Any {
        return answers[groupPosition][childPosition]
    }

    override fun getGroupId(groupPosition: Int): Long {
        return groupPosition.toLong()
    }

    override fun getChildId(groupPosition: Int, childPosition: Int): Long {
        return childPosition.toLong()
    }

    override fun hasStableIds(): Boolean {
        return false
    }

    override fun getGroupView(
        groupPosition: Int,
        isExpanded: Boolean,
        convertView: View?,
        parent: ViewGroup
    ): View? {
        var convertView = convertView
        val question = getGroup(groupPosition) as String
        if (convertView == null) {
            val inflater =
                context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            convertView = inflater.inflate(R.layout.faq_question, null)
        }
        val question2 = convertView?.findViewById<TextView>(R.id.feqTitleView)
        question2?.setTypeface(null, Typeface.BOLD)
        question2?.text = question
        return convertView
    }

    override fun getChildView(
        groupPosition: Int,
        childPosition: Int,
        isLastChild: Boolean,
        convertView: View?,
        parent: ViewGroup
    ): View? {
        var convertView = convertView
        val answer = getChild(groupPosition, childPosition) as String
        if (convertView == null) {
            val inflater =
                context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            convertView = inflater.inflate(R.layout.faq_answer, null)
        }
        val answer2 = convertView?.findViewById<TextView>(R.id.descriptionFaqView)
        answer2?.text = answer
        return convertView
    }

    override fun isChildSelectable(groupPosition: Int, childPosition: Int): Boolean {
        return false
    }

}